File: src/algorithms/quickSort.ts
function quickSort(input: string[]): string[] {
  if (input.length < 2) {
    return input;
  }

  const pivot: string = input[Math.round(input.length / 2)];
  const less: string[] = [];
  const greater: string[] = [];

  for (let i: number = 0; i < input.length; i++) {
    const current: string = input[i];
    (current !== pivot) && ((current < pivot) ? less : greater).push(current);
  }

  return [
    ...quickSort(less),
    pivot,
    ...quickSort(greater)
  ];
}

export default quickSort;

File: grokking-algorithms-python/quicksort.py
# Chapter 4 | Quicksort

def _partition(input, pivot):
    left = []
    right = []
    for el in input:
        if el < input[pivot]:
            left.append(el)
        elif el > input[pivot]:
            right.append(el)
    return left, right

def quicksort(input):
    pivot = round(len(input) / 2)
    if len(input) < 2:
        return input
    left, right = _partition(input, pivot)
    return quicksort(left) + [input[pivot]] + quicksort(right)
